### DISCUSSION TITLE
Strongly signed .dll

### DISCUSSION THREAD

**Discussion by RuoyanMeng** [Apollo3zehn/PureHDF#134] [General]
Hi!
I am currently using the PureHDF library in my project, and I noticed that the current NuGet package does not include strongly-named versions of the DLLs. As my project requires strongly-named assemblies for compatibility with certain features (e.g., global assembly cache, versioning), I wanted to reach out to ask if there are any plans to release a strongly-named version of PureHDF.dll.

Would it be possible for you to publish a version of the package that includes a strongly-named PureHDF.dll? If this isn't currently planned, I would also be open to signing the DLLs myself if that would be helpful in any way.

Thank you very much for your work on this library. I look forward to hearing from you!

**Comment by veltrupdev** [comment]
Hi, thanks for your suggestion. Do you need it for PureHDF 2.x or 1.x?

**Comment by Apollo3zehn** [comment]
Sorry, wrong account. Now its correct :-)

**Comment by Apollo3zehn** [comment]
I think this cannot be done for this project because it is non-profit and so it is too expensive to regularly buy the correct code-signing certificates which are being accepted by Microsoft: https://learn.microsoft.com/en-us/nuget/create-packages/sign-a-package

**Reply by RuoyanMeng** [reply]
Hi, Thanks for your quick reply. I need to strong name .dll for version 1.x. I agree that buying certificate for non-profit project can be difficult. I found this package might be helpful for my case: [strongnamer](https://github.com/[dsplaisted/strongnamer](https://github.com/dsplaisted/strongnamer)) Hope it can help people with similar situation as me.

**Reply by Apollo3zehn** [reply]
This sounds interesting, thanks for sharing that link. Yeah as it is currently implemented by Microsoft, signing DLLs is a bit painful.
